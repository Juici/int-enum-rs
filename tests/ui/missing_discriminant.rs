use int_enum::IntEnum;

#[derive(Clone, Copy, IntEnum)]
#[repr(u8)]
pub enum MissingDiscriminant {
    Zero = 0,
    NoDiscriminant,
}

fn main() {}
