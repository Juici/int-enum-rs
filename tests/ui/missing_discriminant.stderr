error: enum has variants that are not supported by this trait
 --> tests/ui/missing_discriminant.rs:5:10
  |
3 | #[derive(Clone, Copy, IntEnum)]
  |                       ------- in this derive macro expansion
4 | #[repr(u8)]
5 | pub enum MissingDiscriminant {
  |          ^^^^^^^^^^^^^^^^^^^
  |
warning: missing discriminant
 --> tests/ui/missing_discriminant.rs:7:19
  |
7 |     NoDiscriminant,
  |                   ^
  = note: this error originates in the derive macro `IntEnum` (in Nightly builds, run with -Z macro-backtrace for more info)
